server/appservices/oauth2.go

```diff
@@ -23,12 +29,19 @@ func (a *AppServices) StoreOAuth2App(r *incoming.Request, appID apps.AppID, acti
 		return utils.NewUnauthorizedError("%s is not authorized to use remote OAuth2", app.AppID)
 	}
 
+	oldData, _ := json.Marshal(app.RemoteOAuth2)
+	if bytes.Equal(oldData, data) {
+		return nil
+	}
+
 	app.RemoteOAuth2 = oapp
 	err = a.store.App.Save(r, *app)
 	if err != nil {
 		return err
 	}
 
+	a.conf.MattermostAPI().Frontend.PublishWebSocketEvent(config.WebSocketEventRefreshBindings, map[string]interface{}{}, &model.WebsocketBroadcast{})
```

Comment:
This seems quite dangerous. Forcing a bindings refresh on _every_ client could generate a big load. The bindings cache will also miss given that it will depend often on the oauth app.

Reply:
We have no way of refreshing bindings for a specific app :(

For all other installed apps, the bindings would've been cached, but for SN all clients need to refresh

Reply:
Why do we need to refresh the bindings immediately? Client will fresh the bindings soon anyway (channel switch, reconnect event, ...)